src/main/java/htsjdk/samtools/reference/FastaReferenceWriter.java

```diff
@@ -0,0 +1,758 @@
+                                final OutputStream indexOutput,
+                                final OutputStream dictOutput) {
+        this.defaultBasePerLine = checkBasesPerLine(basesPerLine);
+        this.fastaStream = new CountingOutputStream(fastaOutput);
+        this.indexWriter = indexOutput == null ? NullWriter.NULL_WRITER : new OutputStreamWriter(indexOutput, CHARSET);
+        final BufferedWriter dictWriter = new BufferedWriter(dictOutput == null ? NullWriter.NULL_WRITER : new OutputStreamWriter(dictOutput, CHARSET));
+        this.dictWriter = dictWriter;
+        this.dictCodec = new SAMSequenceDictionaryCodec(dictWriter);
+        this.dictCodec.encodeHeaderLine(false);
+        this.sequenceNamesAndSizes = new LinkedHashMap<>();
+    }
+
+    private static Path defaultFaiFile(final boolean makeFaiFile, final Path fastaFile) {
+        return makeFaiFile ? ReferenceSequenceFileFactory.getFastaIndexFileName(fastaFile) : null;
+    }
+
+    private static Path defaultDictFile(final boolean makeDictFile, final Path fastaFile) {
+        return makeDictFile ? ReferenceSequenceFileFactory.getDefaultDictionaryForReferenceSequence(fastaFile) : null;
+    }
+
+    // checks that a sequence name is valid.
+    private static void checkSequenceName(final String name) {
+        ValidationUtils.nonNull(name ,"Sequence name should not be null");
+        ValidationUtils.nonEmpty(name,"Sequence name should not be empty");
+
+        for (int i = 0; i < name.length(); i++) {
+            final char ch = name.charAt(i);
+            if (Character.isWhitespace(ch)) {
+                throw new IllegalArgumentException("the input name contains blank characters: '" + name + "'");
+            } else if (Character.isISOControl(ch)) {
+                throw new IllegalArgumentException("the input name contains control characters: '" + name + "'");
+            }
+        }
+    }
+
+    private static void checkSequenceBases(final byte[] bases, final int offset, final int length) {
+        ValidationUtils.nonNull(bases, "the input bases array cannot be null");
+        final int to = offset + length;
+        for (int i = offset; i < to; i++) {
+            final byte b = bases[i];
+            if (!SequenceUtil.isIUPAC(b)) {
+                throw new IllegalArgumentException("the input sequence contains invalid base calls like: " + (char) b);
+            }
+        }
+    }
+
+    private static String checkDescription(final String description) {
+        if (description == null || description.isEmpty()) {
+            return "";
+        } else {
+            for (int i = 0; i < description.length(); i++) {
+                final char c = description.charAt(i);
+                if (Character.isISOControl(c) && c != '\t') { // tab is the only valid control char in the description.
+                    throw new IllegalArgumentException("the input name contains non-tab control characters: '" +
```

Comment:
The error message says "the input name" but the code refers to this as "description". Which term will be more helpful for the user? It seems like we'd want the user and code names to agree, if possible.

Reply:
it's the name...it's the only part of the description that is user supplied...so that's where weird characters could slip in...

Reply:
Looking only at the file here, there is `sequenceName` which is a parameter to `startSequence()` and `name` which is a parameter to `appendSequence()`. Neither of these are used to create `description`, as far as I can tell. But maybe there's some code further upstream that is accepting a `name` argument which it uses to build `description`?